mdx/guides/use-0x-api-liquidity-in-your-smart-contracts.mdx

```diff
@@ -1,225 +1,199 @@
 
 Begin by cloning the starter project and installing dependencies:
 
-```sh
-$ git clone https://github.com/0xProject/0x-api-starter-guide-code
+```bash
+$ git clone git@github.com:0xProject/0x-api-starter-guide-code
 $ cd 0x-api-starter-guide-code
-$ npm install
+$ npm install --dev
 ```
 
-For development, we will be leveraging the Kovan test network and the respective kovan 0x api endpoint: `kovan.api.0x.org/swap/v1/quote`. 
+### The forked test environment
 
-### Configure the .env file
+All the examples in the project can be run without cost using a forked [ganache](https://github.com/trufflesuite/ganache-cli) instance. To start the forked instance, run the following:
 
+```bash
+# Replace ETHEREUM_RPC_URL with your mainnet node endpoint URL (e.g., infura)
+$ RPC_URL=ETHEREUM_RPC_URL npm run start-fork
 ```
-ETHEREUM_RPC_URL=<INSERT_RPC_URL_HERE>
-MNEMONIC=<INSERT_MNEMONIC_HERE>
-NETWORK_ID=42
-```
-
-For the mnemonic, because we are using the Kovan network, use an account that has some amount of Kovan ETH (KETH). Acquire KETH from a [faucet](https://faucet.kovan.network).
-
-## Fetch a quote from 0x API 
-
-The starter project contains a script that runs web3 interactions with your smart contract, mimicking the behavior of an imaginary DApp front-end. In this example, imagine that a user wants to execute a transaction requiring DAI, but this user only owns ETH. The DApp programmatically swaps ETH for the required DAI and executes the original transaction without any extra input from the user. 
-
-We first use 0x API to get a competitively priced quote for the token swap. By specifying `buyAmount` instead of `sellAmount`, the API understands you intend to buy 1 unit of DAI, not sell 1 unit of Ether.
-
-Open `examples/simple_contract_token_swap.ts` and add the following snippet:
 
-```typescript
-// examples/simple_contract_token_swap.ts
+## The swap contract
 
-// 1. call 0x API for a quote for one dollar of DAI.
-const buyAmount = baseUnitAmount(1); // we want to buy 1 unit of DAI
-const sellAmount = undefined; // we should only specify sellAmount if we want to sell ETH instead of buying DAI
-
-const params = {
-    sellToken: 'ETH',
-    buyToken: 'DAI',
-    buyAmount: buyAmount.toString(),
-    sellAmount: sellAmount.toString(),
-}
-
-const res = await fetch(`https://kovan.api.0x.org/swap/v1/quote?${qs.stringify(params)}`);
-const quote = await res.json();
-
-console.log('Received quote:', quote);
-```
+The contract that will be directly filling 0x-API quotes is [`SimpleTokenSwap`](https://github.com/0xProject/0x-api-starter-guide-code/blob/contracts/SimpleTokenSwap.sol). It implements a `fillQuote()` function that accepts and executes a 0x-API quote to convert some amount of its ERC20 tokens into another. For simplicity, this contract is not designed for use with plain ETH. However, WETH and ETH pairs are identical markets in 0x-API, so this will not affect pricing.
```

Comment:
Heads up, this link currently does work. I guess it should be `https://github.com/0xProject/0x-api-starter-guide-code/blob/v1/contracts/SimpleTokenSwap.sol` or `https://github.com/0xProject/0x-api-starter-guide-code/master/v1/contracts/SimpleTokenSwap.sol` if we merge the code changes.